server/pr_from_issue_handler.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"database/sql"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+func (s *Server) prFromIssueHandler(event *issueEvent, w http.ResponseWriter) {
+	oldPR, err := s.Store.PullRequest().Get(event.Repo.GetOwner().GetLogin(),
+		event.Repo.GetName(),
+		event.Issue.GetNumber())
+	if err != nil {
+		mlog.Error("Error in getting PR from DB", mlog.Err(err))
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// We update the milestone that we have from the issue event and merge it with the PR.
+	// This is necessary to work around caching issues with GitHub.
+	oldPR.MilestoneNumber = sql.NullInt64{Int64: int64(event.Issue.GetMilestone().GetNumber()), Valid: true}
```

Comment:
I got a panic here when I created a PR. (I added it to a milestone before creating it)

Reply:
Ohh .. what a bother. Our code returns nil error _and_ a nil PR if the row is not found .. :angry: